bench_test.go

```diff
@@ -124,9 +125,9 @@ func benchmarkJoin(b *testing.B, format opentracing.BuiltinFormat, numItems int)
 	var carrier interface{}
 	switch format {
 	case opentracing.TextMap:
-		carrier = opentracing.TextMapCarrier{}
+		carrier = opentracing.HTTPHeaderTextMapCarrier{}
 	case opentracing.Binary:
-		carrier = opentracing.BinaryCarrier(&[]byte{})
+		carrier = opentracing.BinaryCarrier(&bytes.Buffer{})
```

Comment:
@bg451 note that we can use bytes.Buffer for a carrier since it's a `io.ReadWriter`